beetmoverscript/script.py

```diff
@@ -86,6 +103,20 @@
             enrich_balrog_manifest(context.release_props, source, locale, destinations)
         )
 
+    context.checksums_manifest.extend(enrich_checksums_manifest(source, pretty_name))
+
+
+def enrich_checksums_manifest(path, pretty_name):
+    digest_info = []
+
+    size = get_size(path)
+    for algo in ("sha512", "md5", "sha1"):
```

Comment:
We seem to be calling `get_size` multiple times as well, although that may be faster than `get_hash`.